pkg/knowledge_base2/properties/bool_property_test.go

```diff
@@ -298,7 +302,19 @@ func Test_BoolPropertyValidate(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			assert := assert.New(t)
 			resource := &construct.Resource{}
-			err := tt.property.Validate(resource, tt.value)
+			graph := construct.NewGraph()
+			for _, r := range tt.testResources {
+				graph.AddVertex(r)
+			}
```

Comment:
nit: Use `graphtest` and `[]any` to make it easier (no tests use this atm though, so it doesn't matter). Same in a couple of the other tests